utils/stall_free.hh

```diff
@@ -178,7 +178,7 @@ future<> clear_gently(foreign_ptr<T>& o) noexcept {
 
 template <typename... T>
 requires (std::is_rvalue_reference_v<T&&> && ...)
-future<> clear_gently(T&&... o) noexcept {
+future<> clear_gently(T&&... o) {
```

Comment:
This should rather be
```patch
- noexcept
+ noexcept(std::is_nothrow_move_constructible<T>::value)
```
or something, no?

Reply:
maybe, but I don't actually see much value in these noexcept subtleties:
* it's used in a [destructor](https://github.com/scylladb/scylladb/blob/master/locator/abstract_replication_strategy.cc#L475), so will abort on exception in any case;
* can throw on task allocation anyway (https://github.com/scylladb/scylladb/pull/13715/files#r1200376871);
* IIUC `boost::icl::interval_map` by default uses `map` under the cover, so it's practically noexcept but don't declare that for some reason.